apollo/drivers/lights.c

```diff
@@ -40,99 +40,93 @@ uint16_t arr[256] = {
 
 void init_Timer(){
 
-    //sets up Timer 2 both channel 1 and 2
+    __HAL_RCC_GPIOA_CLK_ENABLE();
+    __HAL_RCC_GPIOB_CLK_ENABLE();
+
     __HAL_RCC_TIM2_CLK_ENABLE();
-    //sets up the third channel of timer2 by oring bits in the second CCMR register.
     TIM2 -> CCMR2 = (TIM_CCMR2_OC3M_2 | TIM_CCMR2_OC3M_1 | TIM_CCMR2_OC3M_0);
-    //sets up the second channel of timer2. OCxM shows the channel number
-    TIM2 -> CCMR1 = (TIM_CCMR1_OC2M_2 | TIM_CCMR1_OC2M_1 | TIM_CCMR1_OC2M_0);
-    //sets up the capture/compare enable register to output
-    TIM2 -> CCER = (TIM_CCER_CC3E_Msk | TIM_CCER_CC2E_Msk);
-    //sets control register 1 to be enabled
+    TIM2 -> CCMR1 = (TIM_CCMR1_OC2M_2 | TIM_CCMR1_OC2M_1 | TIM_CCMR1_OC2M_0 |
+                     TIM_CCMR1_OC1M_2 | TIM_CCMR1_OC1M_1 | TIM_CCMR1_OC1M_0);
+    TIM2 -> CCER = (TIM_CCER_CC3E_Msk | TIM_CCER_CC2E_Msk | TIM_CCER_CC1E_Msk);
     TIM2 -> CR1 = TIM_CR1_CEN;
-    //initializes the auto-reload register to be 24000
     TIM2 -> ARR = 24000;
-    //initializes capture compare register 2 to be 24000
+    TIM2 -> CCR1 = 24000;
     TIM2 -> CCR2 = 24000;
-    //''            ''      ''       ''    3 '' ''   ''
-    TIM2 -> CCR3 = 24000;
 
-    //intialize GPIOB to be enabled
-    __HAL_RCC_GPIOB_CLK_ENABLE();
+    __HAL_RCC_TIM3_CLK_ENABLE();
+    TIM3 -> CCMR1 = (TIM_CCMR1_OC2M_2 | TIM_CCMR1_OC2M_1 | TIM_CCMR1_OC2M_0) |
+                    (TIM_CCMR1_OC1M_2 | TIM_CCMR1_OC1M_1 | TIM_CCMR1_OC1M_0);
+    TIM3 -> CCER = (TIM_CCER_CC2E_Msk | TIM_CCER_CC1E_Msk);
+    TIM3 -> CR1 = TIM_CR1_CEN;
+    TIM3 -> ARR = 24000;
+    TIM3 -> CCR1 = 24000;
+    TIM3 -> CCR2 = 24000;
+
+    __HAL_RCC_TIM16_CLK_ENABLE();
+    TIM16 -> CCMR1 = (TIM_CCMR1_OC1M_2 | TIM_CCMR1_OC1M_1 | TIM_CCMR1_OC1M_0);
+    TIM16 -> BDTR = TIM_BDTR_MOE_Msk;
+    TIM16 -> CCER = TIM_CCER_CC1E_Msk | TIM_CCER_CC1NE_Msk;
+    TIM16 -> CR1 = TIM_CR1_CEN;
+    TIM16 -> ARR = 24000;
+    TIM16 -> CCR1 = 24000;
+
+    __HAL_RCC_TIM17_CLK_ENABLE();
+    TIM17 -> CCMR1 = (TIM_CCMR1_OC1M_2 | TIM_CCMR1_OC1M_1 | TIM_CCMR1_OC1M_0);
+    TIM17 -> BDTR = TIM_BDTR_MOE_Msk;
+    TIM17 -> CCER = TIM_CCER_CC1E_Msk | TIM_CCER_CC1NE_Msk;
+    TIM17 -> CR1 = TIM_CR1_CEN;
+    TIM17 -> ARR = 24000;
+    TIM17 -> CCR1 = 24000;
+
     GPIO_InitTypeDef GPIO_InitTypeDef;
-    //sets up these pins defined in the header to be outputs
```

Comment:
Is there a reason you removed all the comments?

Reply:
I was moving code around and completely forgot to add them back in, I'll add them back in!